ISSUE-45: Digital Object Full View documentation

mitchellkeaney opened the pull request:
# What
Concise, poetry-less documentation for the Image viewer view mode. 

@DiegoPino I think I need some guidance / input particularly at line 26. It feels necessary to dive into why there are four Strawberrfields there and then what each **FORMATTER** does. But I think if the wording in this section hits a few key points (of which I gave my best shot at there) it would be sufficient for the purpose this documentation is meant to serve.

mitchellkeaney committed:
addressing changes requested in PR #37

this commit includes:
- created new folder in "imgs" folder to contain all documentation screenshots
- edited all screenshots to be jpg and the width equal to 1280px
- revised the changes requested in PR #37 on esmero branch

mitchellkeaney committed:
update to PR #38

- updated folder name
- removed arrow at the end
- updated line 87

mitchellkeaney committed:
select 'photograph' as 'media type'

DiegoPino committed:
Merge pull request #38 from mitchellkeaney/ISSUE-36

#37.a Updates to "First Digital Object" Documentation

DiegoPino committed:
Remove DS_STORE (dot files)

DiegoPino committed:
Remove Old firstobject screenshots

@mitchellkeaney remove the old screenshots was missing.

Given the fact that we will probably repeat a lot of screenshots between docs, it would be useful to have generic Welcome, Add Content, Log in and Logged in screenshots shared by all of them instead of repeating images across folders.

DiegoPino committed:
Remane firstobject screenshot to match pattern

Tiny thing but good to have structure/patterns. I don't really care what the pattern is as long as we enforce it.

DiegoPino committed:
Merge pull request #40 from esmero/ISSUE-36

ISSUE-36: Update First Object Documentation for Beta2

mitchellkeaney committed:
ISSUE-41: How to modify allowed file extensions during ingest (#42)

* first commit: adding md file and all the screenshots for the document

"How to modify allowed file extensions during ingest." Reference to ISSUE-41 in Esmero.

* first solo revision before making pr against esmero

grammar and tense edits

* revisions: new screen shots, less explicit and more contextual text

* grammar and language revisions

* commit 22 April

- consistent file naming p a t t e r n s
- revised the step names
- revised language, added more context at the end of the md to assist in more general help / assistance outside of the guide's specific use case.

* editing grammatical errors

* correct syntax and grammatical errors

- double space between "to" and "fit"; line 9
- double "configure"; line 33
- "Webform"; line 19

mitchellkeaney committed:
Merge pull request #2 from esmero/8.x-1.0-beta2

syncing my beta2 fork

mitchellkeaney committed:
ToC update (#44)

updated the README to include a link to "Customizing Webforms" doc in the table of contents.

mitchellkeaney committed:
Merge pull request #3 from esmero/8.x-1.0-beta2

syncing my branch

mitchellkeaney committed:
Digital Object Full View documentation first pass

- New .md file
- New screenshots
- No inspirational poetry

DiegoPino commented:
Hi Mitchell. Sorry, had to totally remove myself from computer work today.
This is a great read :) Let’s discuss this tomorrow.

My first impression is that this is pretty good! Good and different to what
I thought we discussed in our documentation meeting. This is documenting
the display mode. We discussed we wanted the context, yes, (in that case it
applies!) but we wanted urgently short documentation about our/each field
formatter with their particular settings (mirador, bookreader, sound,
panellum, metadata, etc). If we go for the full Display mode only, which is
fine and you did a great job here, we won’t be able to adapt and reuse docs
if people make change, want to add or configure a different viewer, etc.
Not saying you should revert. Saying this one, as it is, requires extra
info and once its done we need to go back to the each formatter idea. Slow
typing with a single hand so better have a short meeting tomorrow. Thanks a
lot, this is also the right size

DiegoPino commented on docs/digital-object-full-view.md:
```diff
@@ -0,0 +1,32 @@
+# View Modes: Digital Object Full View
+
+`View modes` tell Archipelago Digital Objects (ADOs) how to be displayed to the world. This documentation will give a brief overview of the View mode `Digital Object Full View` and how it can be used when creating or editing ADOs.
```
View Modes (or also know simple as 'Display'). I know it sound super stupid to have two names, but not sure why Drupal has these conflicting names in the menus. This is not a change request, just an open discussion for our call today

DiegoPino replied:
We could make sure people know its a global Drupal thing that affects our ADOs

DiegoPino replied:
... that is provided as a default in your deployment...

DiegoPino commented on docs/digital-object-full-view.md:
```diff
@@ -0,0 +1,32 @@
+`View modes` tell Archipelago Digital Objects (ADOs) how to be displayed to the world. This documentation will give a brief overview of the View mode `Digital Object Full View` and how it can be used when creating or editing ADOs.
+
+## Digital Object Full View
+The `Digital Object Full View` is the default `View mode` used for the Media type `Photograph` and any other Media type that is not already configured to a View mode*. `Digital Object Full View` has the ability to zoom in, zoom out, full screen, rotate and display multiple images associated with an ADO.
```
mapped instead of configured. Here is where the mapping piece needs to be clear given the fact that it only kicks in, if the user does not set one manually during ingest (in other words if the 'default' one is used)

DiegoPino commented on docs/digital-object-full-view.md:
```diff
@@ -0,0 +1,32 @@
+## Digital Object Full View
+The `Digital Object Full View` is the default `View mode` used for the Media type `Photograph` and any other Media type that is not already configured to a View mode*. `Digital Object Full View` has the ability to zoom in, zoom out, full screen, rotate and display multiple images associated with an ADO.
+
+<details><summary>* Configuring ADO Types to View modes </summary>
```
Configuring ADO Types to View modes mapping

DiegoPino commented on docs/digital-object-full-view.md:
```diff
@@ -0,0 +1,32 @@
+<div>
+
+The **ADO Type to View mode Mapping** page can be accessed at */admin/config/archipelago/viewmode_mapping*.
+This page will allow users to configure which `View mode` is used by default via the JSON key `type`.
```
is used by default via the JSON key `type`. i don't have the wording right now, but i feel this needs to be more explicit. Like telling people (inverting the sentence?) that given a certain Media type (which is really any, like ANY JSON key `type` in your metadata) you can setup which existing view mode will be used. And give some example with JSON embeded or such. Do we have docs for this already? I feel i have this somewhere. Also **ADO Type to View mode Mapping** can/should be its own .MD really

DiegoPino commented on docs/digital-object-full-view.md:
```diff
@@ -0,0 +1,32 @@
+![Image Viewer](../imgs/digital-object-full-view/01_imageviewer.jpg)
+
+## Manage display
+In the Webform there is a section titled `DISPLAY SETTINGS` where users will be able to edit which `View mode` is being used for the object. To edit the `View mode`, go to the `Manage display` tab at the top of the Webform.
```
Nope. its not in the Webform.. its in the Ingest form (the webform being only the metadata part). Its part of the Form Mode where the webform is the widget for the SBF and the Display Setting another widget...

DiegoPino commented on docs/digital-object-full-view.md:
```diff
@@ -0,0 +1,32 @@
+
+![Display Settings](../imgs/digital-object-full-view/02_displaysettings.jpg)
+
+First select `Digital Object Full View`, then click on the gear icon under the `OPERATIONS` column to configure the corresponding Strawberryfield. In this case, `🍓Fragola` is the field with the formatter we need for editing this View mode.
```
> is the field with the formatter we need for editing this View mode.
Not clear. Why? You going into edit directly without explaining people why a View mode has a list of all Fields and each one has a formatter.

DiegoPino commented on docs/digital-object-full-view.md:
```diff
@@ -0,0 +1,32 @@
+
+![Manage Display](../imgs/digital-object-full-view/03_managedisplay.jpg)
+
+Inside the Stawberryfield there is the ability to group all media files in a single viewer, select the JSON Key from where to fetch the Media URLs, set the maximum height of the viewer, etc. After any editing, be sure to always click `Update` inside the field and then `Save` at the bottom of the `Manage display` page to persist all changes.
```
This is also not right. This particular formatter (you need to mention its name, and this is where a separate MD file for each formatter and its settings comes handy)

DiegoPino reviewed (changes requested):
Ok, we have stuff to talk about. I feel this just requires some narrowing language and being clear about certain terms. Let's talk about it!

mitchellkeaney committed:
Update to PR #46

Basically created a brand new MD without really meaning to and also added a couple new screenshots.

mitchellkeaney commented:
Hey @DiegoPino ,

The reason the title of the .md document changed is basically I basically started over without actually meaning to. All of the original text I had written I ended up actually changing. Due to those changes, I also decided to rename the document because "digital object full view" is not what this document is actually addressing.
The being said, I feel all these changes to the document addresses the requests made here and also meet the points you and I discussed offline. 

I can think of some spots I will need some fact checking on or ideas for wording, but I think the spine of the document is fully built now.
